mesonbuild/interpreterbase/decorators.py

```diff
@@ -726,16 +741,26 @@ def get_warning_str_prefix(tv: str) -> str:
     def get_notice_str_prefix(tv: str) -> str:
         return 'Future-deprecated features used:'
 
-    def log_usage_warning(self, tv: str, location: T.Optional['mparser.BaseNode']) -> None:
-        args = [
-            'Project targets', f"'{tv}'",
-            'but uses feature deprecated since',
-            f"'{self.feature_version}':",
-            f'{self.feature_name}.',
-        ]
+    def log_usage_warning(self, tv: str, location: T.Optional['mparser.BaseNode'],
+                          always: bool = False) -> None:
+        if always:
+            args = [
+                'Project uses feature',
+                self.feature_name,
+                'which was deprecated in',
+                f"'{self.feature_version}',",
+                'and can safely be replaced without causing a regression.'
+            ]
+        else:
+            args = [
+                'Project targets', f"'{tv}'",
+                'but uses feature deprecated since',
+                f"'{self.feature_version}':",
+                f'{self.feature_name}.',
+            ]
```

Comment:
Not sure why we need 2 different messages here, what's wrong with the original one? Also it probably should be fatal when targeting Meson >=0.63 but since that's inside `project()` I don't think the target meson version is set already, is it?

Sorry for nitpicking, maybe I'm just overthinking this simple PR tbh.

Reply:
I don't think the normal message makes sense in this case, since you'd get a warning like: `Project is targing >= 0.52 but using a feature deprecated since 0.63.` That seems confusing to me.

Reply:
I agree that the dedicated warning message makes sense.

However I also agree with @xclaesse that it seems weird to mark it as non-fatal, because if we always warn about it and can always use the better non-deprecated version of it, then we also do want it to trigger --fatal-meson-warnings.